tests/AstrolabeWorkloadExecutor/Program.cs

```
@@ -14,87 +14,117 @@
             ConsoleCancelEventHandler cancelHandler = (o, e) => HandleCancel(e, cancellationTokenSource);
 
-            var resultsDir = Environment.GetEnvironmentVariable("RESULTS_DIR");
-            var resultsPath = resultsDir == null ? "results.json" : Path.Combine(resultsDir, "results.json");
-            Console.WriteLine($"dotnet main> Results will be written to {resultsPath}...");
-
-            try
-            {
-                Console.CancelKeyPress += cancelHandler;
+            var resultsDir = Environment.GetEnvironmentVariable("RESULTS_DIR") ?? "";
+            var eventsPath = Path.Combine(resultsDir, "events.json");
+            var resultsPath = Path.Combine(resultsDir, "results.json");
+            Console.WriteLine($"dotnet main> Results will be written to {resultsPath}");
+            Console.WriteLine($"dotnet main> Events will be written to {eventsPath}");
 
-                Console.WriteLine("dotnet main> Starting workload executor...");
+            Console.CancelKeyPress += cancelHandler;
 
-                if (!bool.TryParse(Environment.GetEnvironmentVariable("ASYNC"), out bool async))
-                {
-                    async = true;
-                }
+            Console.WriteLine($"dotnet main> Starting workload executor...");
 
-                ExecuteWorkload(connectionString, driverWorkload, async, cancellationTokenSource.Token);
-            }
-            finally
+            if (!bool.TryParse(Environment.GetEnvironmentVariable("ASYNC"), out bool async))
             {
-                Console.CancelKeyPress -= cancelHandler;
-                Console.WriteLine("dotnet main finally> Writing final results file");
-                var resultsJson = ConvertResultsToJson();
-                Console.WriteLine(resultsJson);
-#if NETCOREAPP2_1
-                File.WriteAllTextAsync(resultsPath, resultsJson).Wait();
-#else
-                File.WriteAllText(resultsPath, resultsJson);
-#endif
+                async = true;
```

Why is true the default?

Perhaps we should throw an exception if this required environment variable is not set.

---

it was done previously, but I agreed, it should be updated, done